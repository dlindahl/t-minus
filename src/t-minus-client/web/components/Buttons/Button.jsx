import assign from 'lodash/assign';
import { PropTypes } from 'react';

const baseStyles = {
  root: {
    border: 'none',
    borderRadius: 4,
    cursor: 'pointer',
    fontSize: 15,
    fontWeight: 'normal',
    lineHeight: 1.4,
    marginBottom: 0,
    paddingBottom: 10,
    paddingLeft: 15,
    paddingRight: 15,
    paddingTop: 10,
    textAlign: 'center',
    verticalAlign: 'middle'
  },
  default: {
    background: '#bdc3c7',
    color: '#fff'
  }
};

const Button = (props) => (
  <button onClick={props.onClick} style={assign(baseStyles.root, baseStyles.default)}>
    {props.children}
  </button>
);

Button.propTypes = {
  children: PropTypes.node
};

export default Button;
