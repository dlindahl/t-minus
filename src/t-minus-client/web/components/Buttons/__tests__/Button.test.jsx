import Button from '../Button';
import renderer from 'react-test-renderer';

describe('Beer', () => {
  it('renders the component', () => {
    const component = renderer.create(
      <Button>
        <div/>
      </Button>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('calls the click handler when clicked', () => {
    const onClick = jest.fn();
    const component = renderer.create(
      <Button onClick={onClick}/>
    );
    const tree = component.toJSON();
    tree.props.onClick();
    expect(onClick).toHaveBeenCalled();
  });
});
