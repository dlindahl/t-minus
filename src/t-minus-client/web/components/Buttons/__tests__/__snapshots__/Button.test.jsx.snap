exports[`Beer renders the component 1`] = `
<button
  onClick={undefined}
  style={
    Object {
      "background": "#bdc3c7",
      "border": "none",
      "borderRadius": 4,
      "color": "#fff",
      "cursor": "pointer",
      "fontSize": 15,
      "fontWeight": "normal",
      "lineHeight": 1.4,
      "marginBottom": 0,
      "paddingBottom": 10,
      "paddingLeft": 15,
      "paddingRight": 15,
      "paddingTop": 10,
      "textAlign": "center",
      "verticalAlign": "middle",
    }
  }>
  <div />
</button>
`;
